### PR TITLE
feat(cli): Chokidar-based watch

### PR DESCRIPTION
**Pull request by SevInf**
Supports recursive subfolder watching, which makes it feasible with
`prismaSchemaFolder` and `typedSql`.

Close prisma/team-orm#1256

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -129,5 +130,8 @@
   "dependencies": {
     "@prisma/engines": "workspace:*"
   },
+  "optionalDependencies": {
```
Comment (SevInf) [inline packages/cli/package.json:133]: Chokidar dependency, that we can not bundle (native module).

File: packages/cli/src/generate/Watcher.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  private watcher: FSWatcher
+  private changeQueue = new EventQueue<string>()
+  constructor(rootDir: string) {
+    this.watcher = chokidar.watch(rootDir, {
+      ignoreInitial: true,
+      followSymlinks: true,
+    })
+
+    this.watcher.on('all', (e, path) => {
+      console.log('!!!!')
```
Comment (SevInf) [inline packages/cli/src/generate/Watcher.ts:41]:
```suggestion
```

File: packages/cli/src/generate/Watcher.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  add(watchPath: string) {
+    this.watcher.add(watchPath)
+  }
+
+  async *[Symbol.asyncIterator]() {
+    while (true) {
+      const event = await this.changeQueue.nextEvent()
+      yield event
+    }
+  }
```
Comment (jkomyno) [inline packages/cli/src/generate/Watcher.ts:54]: 👨🏻‍🍳 *chef's kiss*

File: packages/cli/src/generate/Watcher.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  async stop() {
+    await this.watcher.close()
+  }
```
Comment (jkomyno) [inline packages/cli/src/generate/Watcher.ts:58]: Can `stop()` be called on `[Symbol.dispose]() { ... }`?

**Review by jkomyno — APPROVED** [review]

File: packages/cli/src/generate/Watcher.ts (hunk 4 above)
Comment (jkomyno) [reply packages/cli/src/generate/Watcher.ts:58]: (Not needed for this PR, btw)

**Pull request merged**